scripts/quests/outlands/Like_a_Shining_Subligar.lua

```diff
@@ -0,0 +1,108 @@
+local subligarVarName = "shiningSubligar_nb";
+local requiredSubligars = 10
+
+quest.reward =
+{
+    fameArea = xi.quest.fame_area.NORG,
+    fame     = 100,
+    title    = xi.title.LOOKS_SUBLIME_IN_A_SUBLIGAR,
+    item     = xi.items.SCROLL_OF_KURAYAMI_ICHI,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return (status == QUEST_AVAILABLE and player:getFameLevel(xi.quest.fame_area.NORG) >= 3)
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heiji'] = quest:progressEvent(123),
+
+            onEventFinish =
+            {
+                [123] = function(player, csid, option, npc)
+                    quest:begin(player) -- Start Like a Shining Subligar
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heiji'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(124, player:getCharVar(subligarVarName))-- Update the player on the number of Rusty Subligars they have turned in
+                end,
+
+                onTrade = function(player, npc, trade)
+                    local curNumTradedIn = player:getCharVar(subligarVarName)
+                    local numTradingInNow = trade:getItemQty(xi.items.RUSTY_SUBLIGAR);
+                    local remainderNeeded = utils.clamp(requiredSubligars - curNumTradedIn, 0, requiredSubligars);
+
+                    local clampedTrade = math.min(numTradingInNow, remainderNeeded) -- Only take the amount needed to complete the quest and nothing more
+
+                    if npcUtil.tradeHas(trade, { { xi.items.RUSTY_SUBLIGAR, clampedTrade } }) then
+                        local totalWithTrade = curNumTradedIn + clampedTrade
+
+                        if totalWithTrade >= 10 then
+                            return quest:progressEvent(125) -- Complete the quest
+                        elseif totalWithTrade <= 9 then
+                            player:setCharVar(subligarVarName, totalWithTrade) -- Keep track of traded so far
+                            player:confirmTrade()
+                            return quest:event(124, player:getCharVar(subligarVarName))
+                        end
+                    else
+                        player:tradeComplete(false) -- There were extra items in the trade
+                        return quest:event(124, player:getCharVar(subligarVarName))
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [125] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:setCharVar(subligarVarName, requiredSubligars)
+                        player:confirmTrade()
```

Comment:
I am confused on this:

if the quest is completed, what are we confirming for a trade? If its <= 9 then it should also be handled in the eventFinish

Reply:
I was referencing [Recycling Rods](https://github.com/AirSkyBoat/AirSkyBoat/blob/3b961762066e259cbc42c62732ca86c2a6aea8a0/scripts/quests/otherAreas/Recycling_Rods.lua#L4) and following the convention there.

From my understanding a trade needs to be finalized with either a call to confirmTrade or tradeComplete? I can move things around though if need be.